TSRM/tsrm_win32.c

```diff
@@ -662,6 +665,19 @@ TSRM_API int shmget(key_t key, size_t size, int flags)
 		}
 	}
 
+	if (key == IPC_PRIVATE) {
+		/* This should call shm_get with a brand new key id that isn't used yet. See https://man7.org/linux/man-pages/man2/shmget.2.html
+		 * Because shmop_open can be used in userland to attach to shared memory segments, use high positive numbers to avoid accidentally conflicting with userland. */
```

Comment:
I see that sysvshm/shmop PHP modules don't expose this integer id.

It might affect some PECLs if this starts returning negative ids on windows but positive ids on other OSes (e.g. checking `< 0` instead of `-1`), but APCu is at least correct

Reply:
(E.g. for various tutorials that check `< 0` instead of `-1`, negative numbers seem like they'd just cause edge cases or Windows-specific issues)

https://www.csl.mtu.edu/cs4411.ck/www/NOTES/process/shm/shmget.html

```c
shm_id = shmget(IPC_PRIVATE, 4*sizeof(int), IPC_CREAT | 0666);
if (shm_id < 0) {
     printf("shmget error\n");
     exit(1);
}
```

Reply:
> It might affect some PECLs if this starts returning negative ids on windows but positive ids on other OSes (e.g. checking `< 0` instead of `-1`),

Oh, right! [POSIX even mandates](https://pubs.opengroup.org/onlinepubs/9699919799/functions/shmget.html) (emphasis mine):

> Upon successful completion, shmget() shall return a **non-negative** integer, namely a shared memory identifier

So we shouldn't return negative IDs at all (although we do that already if users pass negative keys).

As for the actual range to choose: I'm not sure whether `0x20000000..0x3fffffff` is a good choice. Wrt. `ftok()` `0x00000000..0x1f000000` would make more sense (users would have to pass a control character as `$project_id`), but I assume that such numbers are hard-coded for some existing scripts.